hazelcast/src/main/java/com/hazelcast/jet/pipeline/RemoteMapSourceBuilder.java

```diff
@@ -0,0 +1,176 @@
+ * still give incorrect results without reporting a failure. Concurrent
+ * mutation is not detected at all.
+ * <p>
+ * The default local parallelism for this processor is 1.
+ * <p>
+ * <h4>Predicate/projection class requirements</h4>
+ *
+ * The classes implementing {@code predicate} and {@code projection} need
+ * to be available on the remote cluster's classpath or loaded using
+ * <em>Hazelcast User Code Deployment</em>. It's not enough to add them to
+ * the job classpath in {@link JobConfig}. The same is true for the class
+ * of the objects stored in the map itself. If you cannot meet these
+ * conditions, do not add predicate and projection and add a
+ * subsequent {@link GeneralStage#map map} or {@link GeneralStage#filter
+ * filter} stage.
+ *<p>
+ * @param <K> the type of the key in the map
+ * @param <V> the type of the value in the map
+ * @param <T> the type of the emitted items
+ * @since 5.4
+ */
+public class RemoteMapSourceBuilder<K, V, T> {
+
+    private final String mapName;
+    private String dataConnectionName;
+    private ClientConfig clientConfig;
+    private Predicate<K, V> predicate;
+    private Projection<? super Entry<K, V>, ? extends T> projection;
+
+    RemoteMapSourceBuilder(String mapName) {
+        requireNonNull(mapName, "mapName can not be null");
+        this.mapName = mapName;
+    }
+
+    /**
+     * Set the data connection name to use to connect to the remote cluster.
+     * The data connection must be of
+     * {@link com.hazelcast.dataconnection.HazelcastDataConnection} type.
+     * <p>
+     * One of {@link #dataConnectionName} or {@link #clientConfig} is required
+     * to build the source. If both are provided the data connection takes
+     * precedence.
+     *
+     * @param dataConnectionName name of the data connection
+     * @return this builder
+     */
+    public RemoteMapSourceBuilder<K, V, T> dataConnectionName(@Nonnull String dataConnectionName) {
+        this.dataConnectionName = requireNonNull(dataConnectionName);
+        return this;
+    }
+
+    /**
+     * Set the client configuration to use to connect to the remote cluster.
+     * <p>
+     * One of {@link #dataConnectionName} or {@link #clientConfig} is required
+     * to build the source. If both are provided the data connection takes
+     * precedence.
+     *
+     * @param clientConfig client configuration
+     * @return this builder
+     */
+    public RemoteMapSourceBuilder<K, V, T> clientConfig(@Nonnull ClientConfig clientConfig) {
+        this.clientConfig = requireNonNull(clientConfig);
+        return this;
+    }
+
+    /**
+     * Set the predicate to apply at the source.
+     * See the {@link RemoteMapSourceBuilder} for more details.
+     * @param predicate the predicate
+     * @return this builder
+     */
+    public RemoteMapSourceBuilder<K, V, T> predicate(@Nonnull Predicate<K, V> predicate) {
+        checkSerializable(requireNonNull(predicate), "predicate");
```

Comment:
checkNonNullAndSerializable :)

Reply:
rev : 680737fe0c040b92fd744f238ca4d38b1f8f926f
Performing null check and serializable checks independently with different messages